apis/supernova/src/handler/data-property-handler.js

```diff
@@ -39,54 +123,117 @@ class DataPropertyHandler {
   // ---------------DIMENSION---------------
   // ---------------------------------------
 
+  /**
+   * @returns {Array} Empty array.
+   * @description Returns the default dimension array.
+   * @memberof DataPropertyHandler
+   */
   static getDimensions() {
     return [];
   }
 
+  /**
+   * Gets a dimension by id from dimensions or alternative dimensions.
+   * @param {string} id - The dimension id.
+   * @returns {qix.NxDimension} - The found dimension.
+   * @description Searches for a dimension by id in both main and alternative dimensions.
+   * @memberof DataPropertyHandler
+   * @example
+   * const dim = handler.getDimension('dimId');
+   */
   getDimension(id) {
     const dimensions = this.getDimensions();
     const alternativeDimensions = this.getAlternativeDimensions();
 
     return findFieldById(dimensions, id) ?? findFieldById(alternativeDimensions, id);
   }
 
+  /**
+   * Throws an error indicating the method must be implemented in subclasses.
+   * @throws {Error}
+   * @memberof DataPropertyHandler
+   */
   static getAlternativeDimensions() {
```

Comment:
@Caele I'm not sure what to do with these functions... moving them to nebula.js thrown a typeError so I added `static` to it but don't know if it is how they are supposed to be!

Reply:
hmm, `static` is fine to have on these, but you could also have static on all class functions. Where do you get this typeError?